iceberg/iceberg-handler/src/test/queries/positive/iceberg_optimize_table_partition_evolution_w_dyn_spec_w_filter.q

```diff
@@ -0,0 +1,85 @@
+-- SORT_QUERY_RESULTS
+-- Mask neededVirtualColumns due to non-strict order
+--! qt:replace:/(\s+neededVirtualColumns:\s)(.*)/$1#Masked#/
+-- Mask random uuid
+--! qt:replace:/(\s+uuid\s+)\S+(\s*)/$1#Masked#$2/
+-- Mask a random snapshot id
+--! qt:replace:/(\s+current-snapshot-id\s+)\S+(\s*)/$1#Masked#/
+-- Mask added file size
+--! qt:replace:/(\S\"added-files-size\\\":\\\")(\d+)(\\\")/$1#Masked#$3/
+-- Mask total file size
+--! qt:replace:/(\S\"total-files-size\\\":\\\")(\d+)(\\\")/$1#Masked#$3/
+-- Mask current-snapshot-timestamp-ms
+--! qt:replace:/(\s+current-snapshot-timestamp-ms\s+)\S+(\s*)/$1#Masked#$2/
+--! qt:replace:/(MAJOR\s+succeeded\s+)[a-zA-Z0-9\-\.\s+]+(\s+manual)/$1#Masked#$2/
+-- Mask compaction id as they will be allocated in parallel threads
+--! qt:replace:/^[0-9]/#Masked#/
+-- Mask removed file size
+--! qt:replace:/(\S\"removed-files-size\\\":\\\")(\d+)(\\\")/$1#Masked#$3/
+
+set hive.llap.io.enabled=true;
+set hive.vectorized.execution.enabled=true;
+set hive.optimize.shared.work.merge.ts.schema=true;
+
+create table ice_orc (
```

Comment:
why do we need such a complex test that is just a copy-paste of the existing test? Can't we just use explain?

Reply:
It tests the same logic, but with "optimize table" instead of "alter table" and for that it needs the same preconditions. I wanted to ensure everything is same, so it is easy to compare if it does what it supposed to.

Reply:
it doesn't make sense to execute a duplicate heavy test just for a syntax "sugar"

Reply:
In the test it check e.g. the numFiles changes. Without a table, data, it won't be testable. 
Checking the generated plan would not be the same. To test compaction the best way is:
- create a table
- insert data with multiple insert statement
- check the content of the table
- check table description (e.g. number of files)
- run compaction
- check table content (there must be no change)
- check table description (e.g. number of files decreased)

And this is what this test does. Making it simpler would mean it won't test the whole process.